fs/littlefs/lfs_vfs.c

```diff
@@ -1168,6 +1162,11 @@ static int littlefs_write_block(FAR const struct lfs_config *c,
   FAR struct inode *drv = fs->drv;
   int ret;
 
+  if (fs->readonly)
+    {
+      return -EACCES;
+    }
```

Comment:
are you confident littlefs always handles this error in a way a user usually expects for a read-only mounted filesystem? i think it doesn't. eg. when a write gets delayed to close/sync
i suspect you should reject write attempts in the upper layer. (eg. littlefs_open, littlefs_write etc. or, even fs/vfs.)

also, the usual error for read-only mounted filesystem is EROFS, not EACCES.